tables/cost_and_usage_focus/cost_and_usage_focus_table.go

```diff
@@ -0,0 +1,95 @@
+package cost_and_usage_focus
+
+import (
+	"strings"
+	"time"
+
+	"github.com/rs/xid"
+	typehelpers "github.com/turbot/go-kit/types"
+	"github.com/turbot/pipe-fittings/v2/utils"
+	"github.com/turbot/tailpipe-plugin-aws/sources/s3_bucket"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source"
+	"github.com/turbot/tailpipe-plugin-sdk/artifact_source_config"
+	"github.com/turbot/tailpipe-plugin-sdk/constants"
+	"github.com/turbot/tailpipe-plugin-sdk/row_source"
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+	"github.com/turbot/tailpipe-plugin-sdk/table"
+)
+
+const CostUsageFocusTableIdentifier = "aws_cost_and_usage_focus"
+
+type CostUsageFocusTable struct{}
+
+func (c *CostUsageFocusTable) Identifier() string {
+	return CostUsageFocusTableIdentifier
+}
+
+func (c *CostUsageFocusTable) GetSourceMetadata() []*table.SourceMetadata[*CostUsageFocus] {
+	defaultS3ArtifactConfig := &artifact_source_config.ArtifactSourceConfigImpl{
+		FileLayout: utils.ToStringPointer("%{DATA:prefix}/%{DATA:exportName}/%{DATA:folderName}/%{DATA:billing_period}/%{DATA:assembly_id}/%{DATA}.csv.(?:gz|zip)"),
+	}
+
+	return []*table.SourceMetadata[*CostUsageFocus]{
+		{
+			// S3 artifact source
+			SourceName: s3_bucket.AwsS3BucketSourceIdentifier,
+			Options: []row_source.RowSourceOption{
+				artifact_source.WithDefaultArtifactSourceConfig(defaultS3ArtifactConfig),
+				artifact_source.WithArtifactExtractor(NewCostUsageFocusExtractor()),
+			},
+		},
+		{
+			// any artifact source
+			SourceName: constants.ArtifactSourceIdentifier,
+			Options: []row_source.RowSourceOption{
+				artifact_source.WithArtifactExtractor(NewCostUsageFocusExtractor()),
+			},
+		},
+	}
+}
+
+func (c *CostUsageFocusTable) EnrichRow(row *CostUsageFocus, sourceEnrichmentFields schema.SourceEnrichment) (*CostUsageFocus, error) {
+	row.CommonFields = sourceEnrichmentFields.CommonFields
+
+	// Record standardization
+	row.TpID = xid.New().String()
+	row.TpIngestTimestamp = time.Now()
+
+	// TpIndex
+	switch {
+	case typehelpers.SafeString(row.SubAccountId) != "":
+		row.TpIndex = typehelpers.SafeString(row.SubAccountId)
+	case typehelpers.SafeString(row.BillingAccountId) != "":
+		row.TpIndex = typehelpers.SafeString(row.BillingAccountId)
+	default:
```

Comment:
If we always get the sub/billing account IDs, we should remove this default since we should users an error if we're missing that data, i.e., it was removed manually.

Reply:
Done